### ISSUE TITLE
Only initialize CKAN database if necessary

### ISSUE THREAD

**Issue by torfsen** [oguya/ansible-ckan#12]
Currently, the `paste` command to initialize the CKAN database is always run. As your comment above it already states the initialization should be idempotent, i.e. we should only initialize the database if that hasn't been done before. Your comment suggests touching a file to check this, however I think it's better to simply check whether the database contains any tables.

Here's the code I use to do this. My setup is different enough from yours that I'm currently writing my own playbook (that I might try to merge with yours once I figured it out), but you'll get the principal idea.

```
- name: Check if CKAN database has already been initialized
  command: psql -h {{ db_server }} -U {{ ckan_db_user }} -d {{ ckan_db_name }} -c '\d'
  environment:
    PGPASSWORD: "{{ ckan_db_password }}"
  register: ckan_tables_result

- name: Initialize CKAN database if necessary
  command: '{{ ckan_virtualenv_path }}/bin/paster --plugin=ckan db init -c "{{ ckan_config_path }}/production.ini"'
  when: ckan_tables_result.stdout_lines|length <= 1
```

Basically I use `psql` to execute the command `\d` (list all tables) in the CKAN database. If that returns more than one line then the database is already initialized (`psql` prints a single result line if the database is empty).

**Comment by oguya** [comment]
My original intention was to run `paster` to initialize a db, then touch a file which would prevent ansible from re-initializing ckan db during subsequent runs. But, I think your solution should also work. Go ahead & submit a pull request.
